internal/api/shared/validate/validate_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/perses/perses/internal/api/config"
+	"github.com/perses/perses/internal/api/shared/schemas"
+	modelV1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+const testDataFolder = "testdata"
+
+func TestDashboard(t *testing.T) {
+
+	testSuite := []struct {
+		title            string
+		dashboardFile    string
+		expectedErrorStr string
+	}{
+		{
+			title:            "dashboard with variables queries mixing parent variable & regex",
+			dashboardFile:    "variable_with_regex_dashboard.json",
+			expectedErrorStr: "",
+		},
+	}
+
+	for _, test := range testSuite {
+		t.Run(test.title, func(t *testing.T) {
+			persesDashboardRaw, readErr := os.ReadFile(filepath.Join(testDataFolder, test.dashboardFile))
+			if readErr != nil {
+				t.Fatal(readErr)
+			}
+			schemasService, schErr := schemas.New(config.Schemas{
+				// use the real schemas for these tests
+				VariablesPath: "../../../../schemas/variables",
+				PanelsPath:    "../../../../schemas/panels",
+				QueriesPath:   "../../../../schemas/queries",
```

Comment:
you could do something like that: https://github.com/perses/perses/blob/main/internal/api/e2e/framework/server.go#L66-L69

using `GetRepositoryPath` will avoid to count the number of `../` required :)